gcloud/monitoring/test_client.py

```diff
@@ -157,6 +157,79 @@ def test_metric_descriptor_factory(self):
         self.assertEqual(descriptor.description, DESCRIPTION)
         self.assertEqual(descriptor.display_name, '')
 
+    def test_metric_factory(self):
+        TYPE = 'custom.googleapis.com/my_metric'
+        LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        metric = client.metric(TYPE, LABELS)
+        self.assertEqual(metric.type, TYPE)
+        self.assertEqual(metric.labels, LABELS)
+
+    def test_resource_factory(self):
+        TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        resource = client.resource(TYPE, LABELS)
+        self.assertEqual(resource.type, TYPE)
+        self.assertEqual(resource.labels, LABELS)
+
+    def test_timeseries_factory(self):
+        import datetime
+        from gcloud._testing import _Monkey
+        import gcloud.monitoring.client
+        METRIC_TYPE = 'custom.googleapis.com/my_metric'
+        METRIC_LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        RESOURCE_TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        RESOURCE_LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        VALUE = 42
+
```

Comment:
As far as teaching how things work, the system tests will be a good opportunity for that at the end as well.